src/SpendPermissionManager.sol

```diff
@@ -281,10 +285,16 @@ contract SpendPermissionManager is EIP712 {
 
     /// @notice Revoke a spend permission to disable its use indefinitely.
     ///
-    /// @dev Can only be called by the `account` of a permission.
+    /// @dev Can only be called by the `account` or the `spender` of a spend permission.
     ///
     /// @param spendPermission Details of the spend permission.
-    function revoke(SpendPermission calldata spendPermission) external requireSender(spendPermission.account) {
+    function revoke(SpendPermission calldata spendPermission) external {
+        if (msg.sender != spendPermission.account && msg.sender != spendPermission.spender) {
+            address[] memory allowedSenders = new address[](2);
+            allowedSenders[0] = spendPermission.account;
+            allowedSenders[1] = spendPermission.spender;
+            revert InvalidSender(msg.sender, allowedSenders);
+        }
```

Comment:
Readability is throwing red flags for me. I think I would prefer a separate function dedicated for spender-sent revokes. That way we don't have to change our error, modifier, or this logic. This would reuse the same internal `_revoke` that we are adding anyways for the replace mechanic. I would name the new function something like `spenderRevoke`, `revokeBySpender`, or `reject`.

Reply:
I like `spenderRevoke`